### PR TITLE
Delete old messages when a room reaches 5,000 messages

### PR DESCRIPTION
**Pull request by CS673S17-Team-3**
I implemented this as a receiver that listens for a post_save signal. This means it will check every time a user sends a message to see if that room has reached 5,000 messages. Let me know what you think!

### CODE REVIEW

File: group1/comm/models.py [hunk 1]
```diff
@@ -32,3 +32,13 @@ class UserRoom(models.Model):
...
 	class Meta:
 		unique_together = ('user', 'room')
+
+from django.db.models.signals import post_save
+from django.dispatch import receiver
+
+@receiver(post_save, sender=Message)
+def delete_oldest(sender, instance, **kwargs):
+	curroom = instance.room
+	while (Message.objects.filter(room = curroom).count() > 5000):
```
Comment (onowak-censio) [inline group1/comm/models.py]: this is a really good first step. this query pulls in all room messages into memory, and then issues delete queries on each message after the 5000th. who knows how large a message is ... ? ;) 
it would be great if you could issue one "query" to delete all messages > 5000, rather than pulling in all the room messages into memory and then issuing a delete query per message. 

that would make it cleaner and reduce "load" on the DB (LOL, i know), as well as reduce the memory footprint on the webapp. this is an edge-case, but i think it be a worthwhile exercise learning how to do that. ;)

File: group1/comm/tests.py [hunk 2]
```diff
@@ -1,3 +1,30 @@
...
+
+        User.objects.create(password = "test1", last_login = "2017-02-12 21:36:33.211421", is_superuser = 1, username = "test1", first_name = "", last_name = "", email = "test1@bu.edu", is_staff = 1, is_active = 1, date_joined = "2017-02-12 21:36:33.211421")
+        user = User.objects.get(id = 1);
+
+        #Add 5,100 messages from each room
+        for num in range(0, 5100):
+            Message.objects.create(text = num, time = datetime.now().time(), room = room1, user = user, at_message = False)
+            Message.objects.create(text = num, time = datetime.now().time(), room = room2, user = user, at_message = False)
+
+    def test_count_equal_5000(self):
```
Comment (onowak-censio) [inline group1/comm/tests.py:23]: generally the TestCase "section" is for an "area of concern" in the code (larger than one function). so you might consider having a RoomTestCase "section" instead, where all the Room tests (and setup) can live.

File: group1/comm/tests.py [hunk 3]
```diff
@@ -1,3 +1,30 @@
...
+        room1 = Room.objects.get(name = "test1")
+        room2 = Room.objects.get(name = "test2")
+
+        #Check if each room has 5,000 messages and table has 10,000 overall 
+        self.assertEqual(Message.objects.filter(room = room1).count(), 5000)
+        self.assertEqual(Message.objects.filter(room = room2).count(), 5000)
+        self.assertEqual(Message.objects.count(), 10000)
```
Comment (onowak-censio) [inline group1/comm/tests.py]: these tests are great. keep these tests around, as you refactor your code above. run them while you refactor to make sure your code does what you expect. this is called "Test Driven Development". ;)

Comment (ghost) [reply group1/comm/tests.py]: I second this. The main delight of having unit tests is that if you write them correctly, you can know with absolute certainty that any code-changes you make are non-breaking, at least within the scope of the tests themselves.

**Comment by CS673S17-Team-3** [comment]
This has been updated to be more efficient!

**Comment by oliver-nowak** [comment]
LGTM

File: group1/comm/tests.py [hunk 4]
```diff
@@ -1,3 +1,34 @@
...
+from django.contrib.auth.models import User
 
 # Create your tests here.
+class RoomTestCase(TestCase):
+    def setUp(self):
+    	#Create room and user objects
+        Room.objects.create(name = "test1", description = "test1")
+        room1 = Room.objects.get(name = "test1")
+        Room.objects.create(name = "test2", description = "test2")
+        room2 = Room.objects.get(name = "test2")
```
Comment (ghost) [inline group1/comm/tests.py:13]: More of a question than feedback: why do we need two test rooms here? Can't we just do the assertions on one room? If we have a different test case in which we want to keep track of message counts from multiple rooms then it might be good to implement that in a different test function.

Comment (CS673S17-Team-3) [reply group1/comm/tests.py:13]: Hey Will! I just did this because I was concerned that I was deleting columns from the entire messages table to keep it from exceeding 5000, when what I wanted to do was delete messages that belong to a room that has more than 5000. Maybe I can make it more clear by testing to see if the messages table itself has 10000 messages, like it should.

Comment (oliver-nowak) [reply group1/comm/tests.py:13]: that would be a better test.
